### PR TITLE
Migration to building with Gradle, remove UI and documentation bundles/build

### PR DESCRIPTION
**Pull request by stempler**

### CODE REVIEW

File: buildSrc/src/main/groovy/hale.groovy-conventions.gradle [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ */
+
+plugins {
+  id 'groovy'
+  id 'hale.java-conventions'
+}
+
+spotless {
+  groovy {
+    importOrder()
```
Comment (florianesser) [inline buildSrc/src/main/groovy/hale.groovy-conventions.gradle:12]: While not restoring original behaviour, we could use `importOrder('groovy', 'java', 'javax', 'org', 'com', '')` here to align it with the Java import order I suggested below.

File: buildSrc/src/main/groovy/hale.java-conventions.gradle [hunk 2]
```diff
@@ -6,9 +6,22 @@ plugins {
...
   id 'maven-publish'
 
+  id 'com.diffplug.spotless'
+
   id 'hale.common'
 }
 
+spotless {
+  java {
+    importOrder()
```
[the remark this answers is not in the document]
Comment (stempler) [reply buildSrc/src/main/groovy/hale.java-conventions.gradle:16]: Sure, fine for me.

File: buildSrc/src/main/groovy/hale.groovy-conventions.gradle (hunk 1 above)
Comment (stempler) [reply buildSrc/src/main/groovy/hale.groovy-conventions.gradle:12]: Sure, fine for me.

**Comment by florianesser** [comment]
@stempler I prepared the changes to the `importOrder` configuration and ran `./gradlew spotlessApply` but I can't push the commits. I assume that I don't have write permission to the repository, could you please check?

**Comment by stempler** [comment]
@florianesser Should work now

**Review by florianesser — APPROVED** [review]
LGTM

**Comment by stempler** [comment]
@florianesser I rebased your changes on top of the first time applying the formatting, so we don't have import changes in between. Would merge once the checks are successful.

**Comment by florianesser** [comment]
> This PR cannot be merged yet because a required label is missing: `challenged`. It needs to be added before this PR can be merged.

I added the `challenged` label to this repo now so it can be used for future PRs

**Pull request merged**